src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2691,6 +2757,41 @@ export class StudyViewPageStore
         );
     }
 
+    @action.bound
+    setGenePanelFiltersForChart(meta: ChartMeta, values: string[]): void {
+        const chartUniqueKey = meta.uniqueKey;
+        const suffix = chartUniqueKey.slice(GENE_PANEL_PREFIX.length);
+        const molecularProfileIds = this.molecularProfiles.result
+            .map(mp => mp.molecularProfileId)
+            .filter(mp => mp.endsWith(suffix));
+        const genePanelFiltersForProfiles = values.map(panel => {
+            return {
+                genePanel: panel,
+                molecularProfileIds,
+            };
+        });
+
+        const otherGenePanelFilters = this.genePanelFilters.filter(
```

Comment:
this seems strange.  ```since molecularProfileIds``` is made locally here, i don't think it will EVER be referentially equal to an existing array within genePanelFilters.

Reply:
i.e. even if it has the same items, it won't evaluate equal ```[1,2,3]===[1,2,3]```